devtools/maven/src/main/java/io/quarkus/maven/DependencyTreeMojo.java

```diff
@@ -80,6 +80,7 @@ protected MavenArtifactResolver resolver() throws BootstrapMavenException {
                         .setRemoteRepositoryManager(bootstrapProvider.remoteRepositoryManager())
                         //.setRepositorySystemSession(repoSession) the session should be initialized with the loaded workspace
                         .setRemoteRepositories(repos)
+                        .setPreferPomsFromWorkspace(true)
```

Comment:
This one is not really related to the dependency refactoring. I opened https://github.com/quarkusio/quarkus/pull/18229 for this change.